### PR TITLE
Add pseudolocalization to react-i18n

### PR DESCRIPTION
**Pull request by lemonmade**
Follow up from https://github.com/Shopify/quilt/pull/273. Fixes https://github.com/Shopify/quilt/issues/221.

This PR uses the new pseudotranslation stuff from the above PR in `react-i18n`. Users of that library can now enable pseudotranslation by setting the `pseudolocalize` option on the `Manager` class to `true`, or to a string representing the locale to localize for.

### CODE REVIEW

File: packages/react-i18n/src/i18n.ts [hunk 1]
```diff
@@ -65,17 +78,14 @@ export default class I18n {
...
-  get countryCode(): string | undefined {
-    return this.locale.split('-')[1];
-  }
-
   constructor(
     public translations: TranslationDictionary[],
-    {locale, currency, timezone}: I18nDetails,
+    {locale, currency, timezone, pseudolocalize = false}: I18nDetails,
   ) {
-    this.locale = locale.toLowerCase();
```
Comment (TheMallen) [inline packages/react-i18n/src/i18n.ts]: we don't need to lowercase it anymore?

File: packages/react-i18n/src/i18n.ts [hunk 2]
```diff
@@ -43,12 +44,24 @@ const RTL_LANGUAGES = [
...
+  }
+
+  get region() {
+    return regionFromLocale(this.locale);
+  }
+
+  /**
+   * @deprecated Use I18n#region instead.
+   */
+  get countryCode() {
```
Comment (TheMallen) [inline packages/react-i18n/src/i18n.ts]: this feels a bit weird to do in this PR

File: packages/react-i18n/src/i18n.ts (hunk 1 above)
Comment (lemonmade) [reply packages/react-i18n/src/i18n.ts]: No, this was basically a bug/ incorrect choice in the previous version, an appropriate locale representation has lowercase language and uppercase region

File: packages/react-i18n/src/i18n.ts (hunk 2 above)
Comment (lemonmade) [reply packages/react-i18n/src/i18n.ts]: I definitely see that, but it was a choice between keeping the incorrect language for pseudo localization as well (where we use the region), or updating them both.

**Pull request merged**